grpc.go

```diff
@@ -1180,6 +1202,12 @@ func (a *agentGRPC) CreateSandbox(ctx context.Context, req *pb.CreateSandboxRequ
 	a.sandbox.running = true
 	a.sandbox.sandboxPidNs = req.SandboxPidns
 	a.sandbox.storages = make(map[string]*sandboxStorage)
+	a.sandbox.guestHooks = &specs.Hooks{}
+	a.sandbox.guestHooksPresent = false
+
+	if req.GuestHookPath != "" {
+		a.sandbox.scanGuestHooks(req.GuestHookPath)
```

Comment:
>Another question, since you are relying on the rootfs layout, is there a matching osbuilder PR that allows you to build a rootfs with customized hooks? Or do you simply add the hooks by hand between the rootfs creation and image creation steps?
Currently I am using scripts to add the hooks and other files needed for my use case between rootfs and image creation.

Reply:
I agree with @bergwolf - if a user switches images, I'd say they should expect to have to update the config if one has hooks and the other doesn't.

As an absolute minimum, can you add a log call showing if the user requested hooks *or not* (if `GuestHookPath` is set or not we should log something).

I do wonder if we want to add either a second bool field in the proto file (`bool require_hooks = n`), or maybe even a kernel cmdline option for the agent to consume (the former is cleaner though):

- https://github.com/kata-containers/agent/blob/master/config.go#L67

That way, the agent can clearly establish whether the user wants hooks or not allowing the value (and validity) of `GuestHookPath` to be more clearly determined.

Reply:
@amshinde In that case, I think the `GuestHookPath` config should also be passed by spec (e.g., through labels or annotations).

@eguzman3 I would prefer a fail-early strategy to let users know that they have pointed to a wrong rootfs image. For your mentioned use case, right now users have to remove `GuestHookPath ` if they switch to a rootfs image w/o hooks. In future we might support to specify all kata configs through labels or annotations, and then there is no dependency on a fixed cli configuration and each pod can be customized.

Reply:
ref kata-containers/runtime#753